cpp/tests/c_api/create_sg_graph_test.c

```diff
@@ -0,0 +1,161 @@
+
+  ret_code = cugraph_make_device_buffer(p_handle, vertex_tid, num_edges, &dbuf_dst);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "dst device_buffer creation failed.");
+
+  ret_code = cugraph_make_device_buffer(p_handle, weight_tid, num_edges, &dbuf_wgt);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "weight device_buffer creation failed.");
+
+  ret_code = cugraph_update_device_buffer(p_handle, vertex_tid, &dbuf_src, (byte_t*)h_src);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "src device_buffer update failed.");
+
+  ret_code = cugraph_update_device_buffer(p_handle, vertex_tid, &dbuf_dst, (byte_t*)h_dst);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "dst device_buffer update failed.");
+
+  ret_code = cugraph_update_device_buffer(p_handle, weight_tid, &dbuf_wgt, (byte_t*)h_wgt);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "weight device_buffer update failed.");
+
+  p_graph_envelope = cugraph_make_sg_graph(p_handle,
+                                           vertex_tid,
+                                           edge_tid,
+                                           weight_tid,
+                                           FALSE,
+                                           &dbuf_src,
+                                           &dbuf_dst,
+                                           &dbuf_wgt,
+                                           num_vertices,
+                                           num_edges,
+                                           FALSE,
+                                           FALSE,
+                                           FALSE);
+  runtime_assert(p_graph_envelope != NULL, "graph envelope creation failed.");
+
+  cugraph_free_graph(p_graph_envelope);
+
+  cugraph_free_device_buffer(&dbuf_wgt);
+
+  cugraph_free_device_buffer(&dbuf_dst);
+
+  cugraph_free_device_buffer(&dbuf_src);
+
+  cugraph_free_handle(p_handle);
+
+  return 0;
+}
+
+/*
+ * Since cugraph_make_sg_graph() can return NULL, this ensures
+ * cugraph_free_graph() can accept NULL.
+ */
+int test_free_graph_NULL_ptr()
+{
+  /* Returns void, so check that the call does not crash. */
+  cugraph_free_graph((cugraph_graph_envelope_t*)NULL);
+  return 0;
+}
+
+/*
+ * Test creating a graph with NULL device arrays and "expensive check" enabled.
+ */
+int test_create_sg_graph_bad_arrays()
+{
+  int test_failed = 0;
+
+  cugraph_graph_envelope_t* G = NULL;
+  cugraph_raft_handle_t handle;
+  cugraph_device_buffer_t* src_ptr     = NULL;
+  cugraph_device_buffer_t* dst_ptr     = NULL;
+  cugraph_device_buffer_t* weights_ptr = NULL;
+  size_t num_verts                     = 4;
+  size_t num_edges                     = 3;
+  bool_t do_expensive_check            = 1;
+  bool_t store_transposed              = 0;
+  bool_t is_symmetric                  = 0;
+  bool_t is_multigraph                 = 0;
+
+  /*
+   * FIXME: return success until this test is finished.
+   */
+  return 0;
```

Comment:
Can this test at least be enabled and passing prior to merge?  It seems like this should pass.

Reply:
This has been addressed.